examples/example_plugin/example_plugin/jobs.py

```diff
@@ -85,4 +86,93 @@ def validate_serial(self, serial):
         return False
 
 
-jobs = (ExampleJob, ExampleHiddenJob, ExampleLoggingJob, ExampleJobHookReceiver)
+class ExampleSimpleJobButtonReceiver(JobButtonReceiver):
+    class Meta:
+        name = "Example Simple Job Button Receiver"
+
+    def receive_job_button(self, obj):
+        self.log_info(obj=obj, message="Running Job Button Receiver.")
+        # Add job logic here
+
+
+class ExampleComplexJobButtonReceiver(JobButtonReceiver):
+    class Meta:
+        name = "Example Complex Job Button Receiver"
+
+    def _run_site_job(self, obj):
+        self.log_info(obj=obj, message="Running Site Job Button Receiver.")
+        # Run Site Job function
+
+    def _run_device_job(self, obj):
+        self.log_info(obj=obj, message="Running Device Job Button Receiver.")
+        # Run Device Job function
+
+    def receive_job_button(self, obj):
+        if isinstance(obj, Site):
+            self._run_site_job(obj)
+        elif isinstance(obj, Device):
+            self._run_device_job(obj)
+        else:
+            self.log_failure(obj=obj, message=f"Unable to run Job Button for type {type(obj).__name__}.")
+
+
+class ExampleSiteObjectJob(Job):
+    site = ObjectVar(model=Site)
+
+    class Meta:
+        name = "Example Object Job - Site"
+        description = "I'm an example job that takes a Site object as an input"
+
+    def run(self, data, commit):
+        site = data["site"]
+        if commit:
+            _run_site_job(self, site)
+
+
+class ExampleDeviceObjectJob(Job):
+    device = ObjectVar(model=Device)
+
+    class Meta:
+        name = "Example Object Job - Device"
+        description = "I'm an example job that takes a Device object as an input"
+
+    def run(self, data, commit):
+        device = data["device"]
+        if commit:
+            _run_device_job(self, device)
+
+
+class ExampleJobButtonReceiverOtherJobs(JobButtonReceiver):
+    class Meta:
+        name = "Example Job Button Receiver for Other Jobs"
+
+    def receive_job_button(self, obj):
+        if isinstance(obj, Site):
+            _run_site_job(self, obj)
+        elif isinstance(obj, Device):
+            _run_device_job(self, obj)
+        else:
+            self.log_failure(obj=obj, message=f"Unable to run Job Button for type {type(obj).__name__}.")
+
+
+def _run_site_job(job_class, obj):
+    job_class.log_info(obj=obj, message="Running Site Job.")
+    # Add Site job logic here
+
+
+def _run_device_job(job_class, obj):
+    job_class.log_info(obj=obj, message="Running Device Job.")
+    # Add Device job logic here
```

Comment:
This _might_ be better served as a blog post/user guide on "How to combine a `Job` and `JobButtonReciver`" and might be an advanced use case.

Reply:
Fair.